app/src/main/java/com/humberto/tasky/core/alarm/presentation/AlarmReceiver.kt

```diff
@@ -0,0 +1,74 @@
+package com.humberto.tasky.core.alarm.presentation
+
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.app.TaskStackBuilder
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import androidx.core.app.NotificationCompat
+import androidx.core.net.toUri
+import com.humberto.tasky.R
+import com.humberto.tasky.main.MainActivity
+import dagger.hilt.android.AndroidEntryPoint
+import java.time.LocalDate
+import javax.inject.Inject
+
+@AndroidEntryPoint
+class AlarmReceiver: BroadcastReceiver() {
+
+    @Inject
+    lateinit var notificationManager: NotificationManager
+
+    override fun onReceive(context: Context, intent: Intent) {
+        createNotificationChannel()
+        val id = intent.getStringExtra(ALARM_ID)
+        val title = intent.getStringExtra(ALARM_TITLE)?.takeIf { it.isNotEmpty() } ?: context.getString(R.string.no_title)
+        val description = intent.getStringExtra(ALARM_DESCRIPTION)
+        val itemType = intent.getStringExtra(ITEM_TYPE)
+        val selectedDate = intent.getLongExtra(ITEM_DATE, LocalDate.now().toEpochDay())
+
+        val activityIntent = Intent(context, MainActivity::class.java).apply {
+            data =
+                "tasky://agenda_item/${itemType}?agendaItemId=${id}&isEditing=${0}&selectedDateEpochDay=${selectedDate}"
+            .toUri()
```

Comment:
It shouldn’t be needed to define the names like this. The follow will work:
```
"tasky://agenda_detail/${data.id}/${false}".toUri()
```

for a destination like this:

```
composable<AgendaDetailRoute>(
  deepLinks = listOf(
    navDeepLink<AgendaDetailRoute>(
      basePath = "tasky://agenda_detail”,
    )
  )
)

@Serializable
data class AgendaDetailRoute(
  val id: String,
  val isEditMode: Boolean,
)
```

Reply:
But the only non nullable val I have on my AgendaDetails route is the itemType, that's why I pass them like query params.

Reply:
This table summarises the URI requirements for a Route nicely 😊

|                      | Path parameter                 | Query parameter                                                     |
| -------------------- | ------------------------------ | ------------------------------------------------------------------- |
| **Used for**             | required arguments             | optional arguments<br>collections (arrays and lists) | 
| **Example fields**       | `val id: String`<br>`val code: Int?`  | `val color: String? = null`<br>`val variants: List<String> = emptyList()` |
| **Generated URL format** | `/{id}/{code} `                  | `?color={color}&variants={variant1}&variants={variant2}`|
| **Example URL**          | `product/ABC/123`               | `product?color=red&variants=small&variants=medium` |

Reply:
Also, this URI that I shared was for this model

```
@Serializable
data class AgendaDetailRoute(
  val id: String,
  val isEditMode: Boolean,
)
```

So if yours is different, then the URI will also be.